src/main/java/com/arpnetworking/metrics/mad/Aggregator.java

```diff
@@ -113,8 +114,13 @@ public void notify(final Observable observable, final Object event) {
                     .log();
             return;
         }
+
         final Record record = (Record) event;
-        final Key key = new DefaultKey(createDimensions(record));
+        final ImmutableMap.Builder<String, String> dimensionBuilder = ImmutableMap.builder();
```

Comment:
Agreed.

Since you are adding dimensions to the record anyway, we should map the "legacy" dimensions in each parser into the dimensions map instead of extracting them in the aggregator.